src/pages/view-test-result/view-test-result.ts

```diff
@@ -1,26 +1,122 @@
 import { BasePageComponent } from '../../shared/classes/base-page';
 import { AuthenticationProvider } from '../../providers/authentication/authentication';
+import { SearchProvider } from '../../providers/search/search';
+import { StandardCarTestCATBSchema, ApplicationReference } from '@dvsa/mes-test-schema/categories/B';
+import { tap, catchError } from 'rxjs/operators';
+import { of } from 'rxjs/observable/of';
+import { TestDetailsModel } from './components/test-details/test-details.model';
+import { inflateSync } from 'zlib';
+import { Subscription } from 'rxjs/Subscription';
+import { DateTime } from '../../shared/helpers/date-time';
+import { ExaminerDetailsModel } from './components/examiner-details/examiner-details.model';
 
 @IonicPage()
 @Component({
   selector: 'page-view-test-result',
   templateUrl: 'view-test-result.html',
 })
-export class ViewTestResultPage extends BasePageComponent {
+export class ViewTestResultPage extends BasePageComponent implements OnInit, OnDestroy {
 
   applicationReference: string = '';
 
+  testResult: StandardCarTestCATBSchema;
+  testDetails: TestDetailsModel;
+  examinerDetails: ExaminerDetailsModel;
+
+  isLoading: boolean;
+  loadingSpinner: Loading;
+  subscription: Subscription;
+
   constructor(
     public navController: NavController,
     public platform: Platform,
     public navParams: NavParams,
+    public loadingController: LoadingController,
     public authenticationProvider: AuthenticationProvider,
+    public searchProvider: SearchProvider,
+
   ) {
     super(platform, navController, authenticationProvider);
 
     this.applicationReference = navParams.get('applicationReference');
   }
 
+  ngOnInit(): void {
+    this.handleLoadingUI(true);
+
+    this.subscription = this.searchProvider
+      .getTestResult(this.applicationReference, this.authenticationProvider.getEmployeeId())
+      .pipe(
+        tap((data) => {
+          const buffer = Buffer.from(data, 'base64');
+          const decompressedData = inflateSync(buffer).toString();
+          this.testResult = JSON.parse(decompressedData) as StandardCarTestCATBSchema;
+          this.generateData();
+          this.handleLoadingUI(false);
+        }),
+        catchError((error) => {
+          console.log('ERROR', JSON.stringify(error));
+          return of();
+        }),
+      ).subscribe();
+  }
+
+  ngOnDestroy(): void {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  handleLoadingUI = (isLoading: boolean): void => {
+    this.isLoading = isLoading;
+
+    if (isLoading) {
+      this.loadingSpinner = this.loadingController.create({
+        dismissOnPageChange: true,
+        spinner: 'circles',
+      });
+      this.loadingSpinner.present();
+      return;
+    }
+    if (this.loadingSpinner) {
+      this.loadingSpinner.dismiss();
+      this.loadingSpinner = null;
+    }
+  }
+
+  generateData(): void {
```

Comment:
Is this generating data or just formatting/structuring it?

Reply:
Formatting will rename :)